coordinator/utils/utils.py

```diff
@@ -210,6 +211,18 @@ def merge_coverage_files(data_files, output_name, ftype='branch-only'):
         error_msg("can not merge coverage file {0}".format(output_name))
         return False
 
+##### PR: filename mismatch
+def from_simple_to_afl_name(simple_path):
+    tmp = os.path.basename(simple_path).replace("_",":")
+    tmp = glob.glob(os.path.join(os.path.dirname(simple_path), tmp) + "*")
+    if not tmp:
+        return ""
+    else:
+        return tmp[0]
```

Comment:
ah never mind, glob.glob returns a list,
can we add an assert here to ensure the list len  is 1?

Reply:
I think an assert is too strong since a file can be removed by AFL on the fly.
Time to time AFL calls a routine to polish the queue (a [cmin](https://github.com/mirrorer/afl/blob/master/afl-cmin) similar function if you want), this is briefly mentionned [here](https://github.com/mirrorer/afl/blob/2fb5a3482ec27b593c57258baae7089ebdc89043/docs/README#L71) as a part of afl-fuzz algorithm.
Unfortunately, it may raise the assertion if the file savior wants to read has been removed by AFL.
I preferred the way you chose [here](https://github.com/evanmak/savior-source/blob/0991fbbda4c92589d26f768978005d9c9e7a9ff2/coordinator/SEs/klee_sym_explorer.py#L117) and simply continue if a problem occurred.

Reply:
> > The problem are the first_seen values in AFL. You cannot retrieve easily filenames from ids as it is currently implemented in AFL. This is the reason why I kept simple file format inside the coordinator, demanding the format conversions.
> 
> it's been a few years since i work on this code, can you elaborate a bit more, why we can't keep the naming scheme consistent (i.e., use the standard naming across all modules?), the first_seen values in AFL can be modified here:
> 
> https://github.com/evanmak/savior-source/blob/b419c868e3dd966e77bb2707a7901e6f1872fb92/AFL/afl-fuzz.c#L3546
> 
> would it help make things come cleaner?

The problem here is the use of `edge_san_first_seen[i]` storing only the id of the first testcase covering a branch. I have not seen a simple way to print back the full filename in the standard format. A solution would be to store the full name but it does not sound like a simpler way.

Reply:
Hmmm, I got your point, reading the code again we use the seed names in the `input_id_map` for SE converter, so it needs to be a full match.

Thanks for the discussion btw, my concern was keeping a mixed scheme will make the code logic more convoluted, being able to modified KLEE seems like a more straightforward approach but we don't have source. 

@DanielGuoVT maybe you could consider release another klee version before fully open source. But until then we can use the solution in this PR.